shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -149,11 +149,35 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
   } else if ([method isEqualToString:@"Share.invoke"]) {
     [self showShareViewController:args];
     result(nil);
+  } else if ([method isEqualToString:@"ContextMenu.showSystemContextMenu"]) {
+    [self showSystemContextMenu:args];
+    result(nil);
+  } else if ([method isEqualToString:@"ContextMenu.hideSystemContextMenu"]) {
+    [self hideSystemContextMenu];
+    result(nil);
   } else {
     result(FlutterMethodNotImplemented);
   }
 }
 
+- (void)showSystemContextMenu:(NSDictionary*)args {
+  if (@available(iOS 16.0, *)) {
+    FlutterTextInputPlugin* textInputPlugin = [_engine.get() textInputPlugin];
+    BOOL shownEditMenu = [textInputPlugin showEditMenu:args];
+    if (!shownEditMenu) {
+      FML_LOG(ERROR) << "Only text input supports system context menu for now. See "
+                        "https://github.com/flutter/flutter/issues/143033.";
```

Comment:
@justinmc added this warning if fails to show edit menu (e.g. when calling from non-text input).

Reply:
Could you add this sentence if it's not too long?

"Ensure the system context menu is shown with an active text input connection."

That way it's more clear what someone needs to do to fix this, if it wasn't already obvious.